### ISSUE TITLE
[a11y]: Carbon DataTable Accessibility Issue: "Element "button" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'"

### ISSUE THREAD

**Issue by AlmondBro** [carbon-design-system/carbon#19270]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.76.0

### React version

v18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

Hi Carbon team,

As part of an IBM-wide requirement, a requirement is for DataStage UI (DataStage/ds-canvas) to be accessibility compliant, via the [IBM Equal Access Accessibility Checker tool](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp?hl=en-US).

### Reproduction/example

https://stackblitz.com/edit/github-voohh6bt-xxinzf67?file=src%2FApp.jsx

### Steps to reproduce

1. Open StackBlitz URL.

2. Open & execute the **IBM Equal Access Accessibility Checker Tool**

See below violation:

> Element "button" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true

![Image](https://github.com/user-attachments/assets/09c6596e-6de0-4493-9f95-da2ad3acd7cf)

![Image](https://github.com/user-attachments/assets/de13a191-7c54-4a57-80c6-8fc4e717b92d)

![Image](https://github.com/user-attachments/assets/f211bd03-a5d5-4b0e-a4e8-81963bc0939c)

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey , Can not reproduce this violation on  the shared stackblitz and on @carbon/react@1.84.0 (latest version)
Happy to reopen if you still see this issue
